### PR TITLE
expose groupfolder through dav endpoint

### PR DESCRIPTION
**Pull request by icewind1991**
expose groupfolders at `/remote.php/dav/groupfolders/<user id>`

you can also request the mount point for the group folder by requesting the `{http://nextcloud.org/ns}mount-point` property

- [x]  https://github.com/nextcloud/server/pull/37824

### CODE REVIEW

**Comment by skjnldsv** [comment]
> you can also request the mount point for the group folder by requesting the `{http://nextcloud.org/ns}mount-point` property

The return is odd. Given a `Group folder` share, the mountpoint is `/admin/files/Group folder`, regarding dav, shouldn't it be `/files/admin/Group folder` instead?
What I need is the original files location `$node->getFileInfo()->getPath()`, do you want me to change the mountpoint or add a new prop ?

**Comment by icewind1991** [comment]
> > you can also request the mount point for the group folder by requesting the `{http://nextcloud.org/ns}mount-point` property
> 
> The return is odd. Given a `Group folder` share, the mountpoint is `/admin/files/Group folder`, regarding dav, shouldn't it be `/files/admin/Group folder` instead? What I need is the original files location `$node->getFileInfo()->getPath()`, do you want me to change the mountpoint or add a new prop ?

This is the difference between "dav paths" and "filesystem paths". I've changed it to return the path relative to the user home directory (`/Group folder`)

File: src/actions/openGroupfolderAction.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ */
+import { translate as t } from '@nextcloud/l10n'
+import type { Node } from '@nextcloud/files'
+
+import { registerFileAction, FileAction } from '@nextcloud/files'
+import { getCurrentUser } from '@nextcloud/auth'
+
+registerFileAction(new FileAction({
+	id: 'opengroupfolders',
```
Comment (artonge) [inline src/actions/openGroupfolderAction.ts:29]: iwouldusesnakecase

**Comment by skjnldsv** [comment]
Ready for my part :+1: 
devcom_apps_files_groupfolders_dir=_

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
@artonge @icewind1991 if you have some time to quickly test the front, would be nice! :)

**Review by artonge — COMMENTED** [review]
- [ ] If no actions available, should we allow selection?
- [ ] Is there a way to provide icons to be displayed on the folder or file icon to differentiate groupfolder, share, e2ee, ...
- [ ] The home icon is useless, can it be removed?

File: src/services/client.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { createClient } from 'webdav'
+import { generateRemoteUrl } from '@nextcloud/router'
+import { getCurrentUser, getRequestToken } from '@nextcloud/auth'
+
+export const rootPath = `/groupfolders/${getCurrentUser()?.uid}`
+export const rootUrl = generateRemoteUrl('dav' + rootPath)
+const client = createClient(rootUrl, {
+	headers: {
+		requesttoken: getRequestToken() || '',
+	},
```
[the remark this answers is not in the document]
Comment (skjnldsv) [reply src/services/client.ts:31]: Will be tackled in server, this is part of the "to investigate" list :)

**Comment by skjnldsv** [comment]
> *  If no actions available, should we allow selection?

For server

> *  Is there a way to provide icons to be displayed on the folder or file icon to differentiate groupfolder, share, e2ee, ...

Will be, also for server. The old way was a bit hacky, I will provide a dedicated MimeIcon API to register inline svg 

> * The home icon is useless, can it be removed?

It's not, clicking the last breadcrumb refreshes the current folder.
Here it would refresh the list of groupfolders (as in legacy files)
Nonetheless, this would be an issue for server and in the future, there is no doubt the groupfolder will not be just a redirect anymore but fully browsable. Let's keep it like that :)

**Review by artonge — APPROVED** [review]

**Pull request merged**